### PR TITLE
Pass correct config to standalone

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Passes the correct config file into local mode.
PR https://github.com/elastic/beats/pull/27118 changed how path is retrieved in `run.go`
and went from 
`pathConfigFile := paths.ConfigFile()`
to 
`pathConfigFile := paths.AgentConfigFile()`

while `ConfigFile()` checks for config file defined using `-c` and if not found it falls back to `elastic-agent.yml` 
`AgentConfigFile()` just retrieves an absolute path to `fleet.yml`

in this PR i pass result of `paths.ConfigFile()` into standalone code path so it behaves like it was before.

As standalone agent needs to read `elastic-agent.yml` because this is where the config is, `fleet.yml` is still created and it contains agent.id 

Possible workaround is to COPY config into `fleet.yml`. renaming will lead to agent failing on initial config load. 
Then agent will append agent id into `fleet.yml` while preserving inputs and outputs. Just tested it.
 
## Why is it important?

WIthout this standalone does not work as it tries to read fleet.yml 

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/application/application.go
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/application/application.go:76]: Can you share a bit more in the PR description on why pathConfigFile contains the wrong info on how this change fixes it?

**Review by kvch — COMMENTED** [review]
Could you please add some tests to check if the application is initialized correctly in the appropriate mode and options? IIRC it is not the first time that a standalone agent had such issues.

**Comment by michalpristas** [comment]
@ruflin @kvch @blakerouse i added details into a description

**Comment by michalpristas** [comment]
@kvch we have e2e tests but it seems we dont' run them

**Comment by kvch** [comment]
I assumed we have some e2e tests. I would rather look for something lightweight that can tell you in a few seconds that something broke.

**Review by blakerouse — APPROVED** [review]
Good catch.

I would have expected that even if e2e is not run on every change, that it is run once a day from 7.x and master. So why was this not found and reported? Nothing that needs to block this PR, but a valid question.

This change is not in 7.14, the original PR only backported to 7.x (aka. 7.15).

**Comment by ruflin** [comment]
@michalpristas @blakerouse Can one of you follow up on this to make sure this is covered by the e2e test? We should also be able to cover this directly with some tests in the repo itself?

**Comment by mdelapenya** [comment]
> @kvch we have e2e tests but it seems we dont' run them, standalone scenario must be failing with this

the e2e tests run nightly for 7.14 and they did detect the failure. For PRs, it's still possible to run them ideally **before** a PR is merged. Just add the `/package` comment in the PR so that the packaging (which is mandatory to run the e2e tests against the produced binaries) is executed.

OTOH I agree with @kvch, if possible a unit test would be super

**Comment by mdelapenya** [comment]
@blakerouse The e2e tests did catch it: https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2F7.14.x/detail/7.14.x/212/tests. All stand-alone scenarios are failing. The nightly build did not catch it because it uses released artifacts from Elastic's artifactory, not the binaries produced by Beats CI. The above CI link represents the execution of a merge commit on Beats.

**Comment by mdelapenya** [comment]
Should this fix be backported to 7.14?

**Pull request merged**
